analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -89,33 +89,45 @@ private IEnumerable<ExplodedNode> ProcessBranching(ExplodedNode node)
             }
             else if (node.Block.ContainsThrow())
             {
-                yield return CreateNode(cfg.ExitBlock, null, null);
+                yield return new(cfg.ExitBlock, node.State, null);
             }
             else
             {
                 foreach (var successor in node.Block.Successors.Where(x => IsReachable(node, x)))
                 {
-                    if (successor.Destination is not null)
+                    if (ProcessBranch(node, successor) is { } newNode)
                     {
-                        yield return successor.FinallyRegions.Any() // When exiting finally region(s), redirect to 1st finally instead of the normal destination
-                            ? FromFinally(successor, new FinallyPoint(node.FinallyPoint, successor))
-                            : CreateNode(successor.Destination, successor, node.FinallyPoint);
-                    }
-                    else if (successor.Source.EnclosingRegion.Kind == ControlFlowRegionKind.Finally)    // Redirect from finally back to the original place (or outer finally on the same branch)
-                    {
-                        yield return FromFinally(successor, node.FinallyPoint.CreateNext());
+                        yield return newNode;
                     }
                 }
             }
+        }
+
+        private ExplodedNode ProcessBranch(ExplodedNode node, ControlFlowBranch branch)
+        {
+            if (branch.Destination is not null)
+            {
+                return branch.FinallyRegions.Any() // When exiting finally region(s), redirect to 1st finally instead of the normal destination
+                    ? FromFinally(new FinallyPoint(node.FinallyPoint, branch))
+                    : CreateNode(branch.Destination, node.FinallyPoint);
+            }
+            else if (branch.Source.EnclosingRegion.Kind == ControlFlowRegionKind.Finally)    // Redirect from finally back to the original place (or outer finally on the same branch)
+            {
+                return FromFinally(node.FinallyPoint.CreateNext());
+            }
+            else
+            {
+                return null;    // We don't know where to continue
```

Comment:
This is not coverable. It could end up here with rethrow, but we don't explore `catch` so it's not reachable.